src/components/router/Router.brs

```diff
@@ -51,22 +48,45 @@ sub resetHistory(rootPath = "" as String)
   m._history = []
 
   if (rootPath <> "")
-    m._history.push({ path: rootPath, params: {} })
+    rootRoute = CreateObject("roSGNode", "ActivatedRoute")
+    rootRoute.setFields({ path: rootPath })
+
+    m._history.push(rootRoute.getFields())
   end if
 end sub
 
-sub _updateHistory()
-  if (m.top.url = "" OR m.top.activatedRoute.shouldSkip)
-    return
+function _getPreviousRoute(data as Object) as Object
+  isBackJourney = getProperty(data, "isBackJourney", false)
+
+  if (isBackJourney OR m.top.activatedRoute.shouldSkip)
+    previousRoute = CreateObject("roSGNode", "ActivatedRoute")
+    previousRouteData = m._history[m._history.count() - 1]
+
+    if (previousRouteData = Invalid) then return Invalid
+
+    previousRoute.setFields(previousRouteData)
+
+    return previousRoute
   end if
 
+  return NodeUtils().cloneNode(m.top.activatedRoute)
+end function
+
+sub _updateHistory()
+  if (m.top.url = "" OR m.top.activatedRoute.shouldSkip) then return
+
   m._history.push(_createHistoryItem(m.top.activatedRoute))
 end sub
 
 function _createHistoryItem(route as Object) as Object
-  return {
-    path: route.path,
-    params: route.params,
-    backJourneyData: route.backJourneyData,
-  }
+  historyItem = route.getFields()
+  _deleteKeys(historyItem, ["change", "focusable", "focusedChild", "id"])
+
+  return historyItem
```

Comment:
having the `omit` function added to the `ObjectUtils` we could simply do here as per below
```suggestion
  return ObjectUtils().omit(route.getFields(), ["change", "focusable", "focusedChild", "id"])
```